framework/source/class/qx/io/remote/Rpc.js

```diff
@@ -466,34 +523,42 @@ qx.Class.define("qx.io.remote.Rpc",
           };
 
         // Only add the server_data member if there is actually server data
-        if (serverData)
-        {
+        if (serverData) {
           requestObject.server_data = serverData;
         }
-      }
-      else
-      {
+      } else {
+        // JSONRPC v2
         // If there's a service name, we'll prepend it to the method name
         service = this.getServiceName();
-        if (service && service != "")
-        {
+        if (service && service != "") {
           service += ".";
-        }
-        else
-        {
+        } else {
           service = "";
         }
 
         // Create a standard version 2.0 rpc data object
-        requestObject =
-          {
-            "jsonrpc" : "2.0",
-            "method"  : service + method,
-            "id"      : id,
-            "params" : parameters
-          };
+        // method and jsonrpc are mandatory
+        requestObject = {
+          "jsonrpc" : "2.0",
+          "method"  : service + method
+        };
+        // parameters are optional according to the spec
+        if (parameters) {
+          if (qx.lang.Type.isObject(parameters) || qx.lang.Type.isArray(parameters)) {
+            requestObject["params"] = parameters
+          } else {
+            throw new Error("Parameters must be type Array (positional) or Object (named)");
+          }
+        }
+        // id: null is allowed but discouraged
```

Comment:
The spec says this:
-  "A Notification is a Request object without an "id" member. "
- "The use of Null as a value for the id member in a Request object is discouraged, because this specification uses a value of Null for Responses with an unknown id. Also, because JSON-RPC 1.0 uses an id value of Null for Notifications this could cause confusion in handling."